packages/teleport/src/Nodes/NodeAdd/NodeAddByManual.tsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2020 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Copyright 2020 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import useTeleport from 'teleport/useTeleport';
+import cfg from 'teleport/config';
+import { Text, ButtonLink, Box } from 'design';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+
+export default function NodeAddByManaul(props: Props) {
+  const linux64Link = getDownloadLink('linux64');
+  const linux32Link = getDownloadLink('linux32');
+  const macLink = getDownloadLink('mac');
+
+  return (
+    <>
+      <Box {...props}>
+        <Text>Step 1: Download and Install tctl</Text>
+        <ButtonLink href={macLink} target="_blank">
+          Mac
+        </ButtonLink>{' '}
+        |
+        <ButtonLink href={linux64Link} target="_blank">
+          Linux 64-bit
+        </ButtonLink>{' '}
+        |
+        <ButtonLink href={linux32Link} target="_blank">
+          Linux 32-bit
+        </ButtonLink>
+      </Box>
+      <Box>
+        <Text>Step 2: Generate a new node invite token</Text>
+        <TextSelectCopy text="tctl nodes add" isTerminal={true} />
+      </Box>
+    </>
+  );
+}
+
+function getDownloadLink(type: 'mac' | 'linux32' | 'linux64') {
+  const version = useTeleport().storeUser.state.cluster.authVersion;
```

Comment:
normally you would use hooks at the beginning of the component function. 
getDownloadLink is not a component.

Pass the version to `ByManaul.tsx` via prop. This simple component does not need to depend on TeleportContext, TeleportContextProvider, useTeleport. and config.ts

Reply:
revised